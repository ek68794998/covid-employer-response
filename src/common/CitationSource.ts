export class CitationSource {
	public date?: string;

	public link: string = "";

	public source: string = "";

	public title: string = "";
}
