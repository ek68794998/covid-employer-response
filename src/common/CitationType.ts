export const CitationTypeValues: string[] = [
	"hearsay",
	"statement",
	"publication",
];

export type CitationType = typeof CitationTypeValues[number];

export const getCitationTypeValue = (type?: CitationType): number =>
	type ? CitationTypeValues.indexOf(type) : -1;
