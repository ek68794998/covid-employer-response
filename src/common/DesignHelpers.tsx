import React from "react";

export class DesignHelpers {
	public static materialIcon(iconName: string, className?: string): JSX.Element {
		return <i className={`material-icons ${className || ""}`}>{iconName}</i>;
	}
}
