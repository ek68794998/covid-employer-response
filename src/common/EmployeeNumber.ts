type EmployeeMagnitude = 6 | 9 | null;

export class EmployeeNumber {
	public exactValue: number;

	public magnitude: EmployeeMagnitude;

	public value: number;

	public constructor(value: number) {
		this.exactValue = value;

		this.magnitude = EmployeeNumber.getEmployeeMagnitude(value);

		this.value =
			this.magnitude
				? Math.round(1234567 / Math.pow(10, this.magnitude - 1)) / 10
				: value;
	}

	public get magnitudeLabel(): string | null {
		return this.magnitude === 6 ? "million" : this.magnitude === 9 ? "billion" : null;
	}

	public toString(): string {
		return this.magnitude ? `${this.value} ${this.getEmployeeMagnitudeName()}` : this.value.toLocaleString();
	}

	private static getEmployeeMagnitude(value: number): EmployeeMagnitude {
		if (!value) {
			return null;
		}

		if (value >= 1000000000) {
			return 6;
		}

		if (value >= 1000000) {
			return 9;
		}

		return null;
	}

	private getEmployeeMagnitudeName(): string | null {
		switch (this.magnitude) {
			case 6: return "million";
			case 9: return "billion";
			default: return null;
		}
	}
}