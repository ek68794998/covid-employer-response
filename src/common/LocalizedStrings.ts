export interface LocalizedStrings { [key: string]: LocalizedStringValue }

export const format = (translation: string, args: { [key: string]: any }): string => {
	if (!translation) {
		throw new Error("Translated string must not be empty.");
	}

	return translation.replace(/\{([^\}]+)\}/g, (value: string, argName: string) => args[argName] || argName);
};

// Would prefer to use 'string | { [key: string]: LocalizedStringValue }' here.
// However, then TypeScript complains if you do 'strings.foo.bar', saying that 'bar' does not exist on type 'string'.
type LocalizedStringValue = any;