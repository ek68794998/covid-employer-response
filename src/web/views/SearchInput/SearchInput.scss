@import "../Colors.scss";

$input-buffer: 8px;
$input-height: 40px;

div.SearchInput__Container {
	align-items: center;
	border: 1px solid $outline-color;
	border-radius: ($input-height / 2);
	display: flex;
	height: $input-height;
	padding: 0 ($input-height / 2 - $input-buffer) 0 $input-buffer;
}

input[type=search] {
	background: none;
	border: 0;
	font: 400 16px 'Baloo Chettan 2';
	outline: 0;
	width: 100%;
}