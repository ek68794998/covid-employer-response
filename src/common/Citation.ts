import { CitationSource } from "./CitationSource";
import { CitationType } from "./CitationType";

export class Citation {
	public positivity: number = 0;

	public sources?: CitationSource[];

	public summary: string = "";

	public type?: CitationType;
}
