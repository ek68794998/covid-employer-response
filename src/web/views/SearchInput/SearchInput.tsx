import React from "react";
import { useSelector } from "react-redux";
import { RouteProps } from "react-router-dom";

import { LocalizedStrings } from "../../../common/LocalizedStrings";

import { AppState } from "../../state/AppState";
import { getStrings } from "../../state/ducks/localization/selectors";

import "./SearchInput.scss";

interface Props extends RouteProps {
	onChange: (value: string) => void;
}

const SearchInput: React.FC<Props> = (props: Props): React.ReactElement => {
	const strings: LocalizedStrings = useSelector((state: AppState) => getStrings(state));
	const { onChange } = props;

	return (
		<div className="search-input-container">
			<i className="material-icons">search</i>
			<input
				onInput={(e: React.FormEvent<HTMLInputElement>): void => onChange(e.currentTarget.value)}
				placeholder={strings.search}
				type="search"
			/>
		</div>
	);
};

export default SearchInput as any;
