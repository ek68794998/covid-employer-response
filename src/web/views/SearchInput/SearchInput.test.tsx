import React from "react";
import { render } from "react-dom";

import { LocalizedStrings } from "../../../common/LocalizedStrings";

import SearchInput from "./SearchInput";

jest.mock("react-redux", () => ({
	useSelector: (): {} => ({
		getStrings: (): LocalizedStrings => ({ search: "Search" }),
	}),
}));

describe("<SearchInput />", () => {
	test("renders without exploding", () => {
		const div: HTMLDivElement = document.createElement("div");

		render(
			<SearchInput.WrappedComponent />,
			div,
		);
	});
});
