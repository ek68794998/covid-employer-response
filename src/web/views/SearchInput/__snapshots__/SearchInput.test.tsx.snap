// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> renders without exploding 1`] = `
<div
  className="SearchInput__Container"
>
  <i
    className="material-icons"
  >
    search
  </i>
  <input
    onInput={[Function]}
    placeholder="Search"
    type="search"
  />
</div>
`;
